core/data.go

```diff
@@ -505,6 +505,43 @@ type PaymentMetadata struct {
 	CumulativePayment *big.Int `json:"cumulative_payment"`
 }
 
+func NewPaymentMetadata(
```

Comment:
personally not a big fan of having a bunch of miscellaneous stuff under `core.`.
Do we really need this here or can we also keep it under `core/payments`?